File: Typescript/Bot.ts
const TelegramBot = require('node-telegram-bot-api');

const main = () => {
  const token = process.env.TELEGRAM_BOT_TOKEN;

  if (!token) {
    console.log('Error: TELEGRAM_BOT_TOKEN is not set.');
    return;
  }

  const bot = new TelegramBot(token, { polling: true });

  bot.onText(/^\/start$/, (msg) => {
    bot.sendMessage(
      msg.chat.id,
      'Hello! I am your simple bot.\nUse /help to see commands.',
    );
  });

  bot.onText(/^\/help$/, (msg) => {
    bot.sendMessage(
      msg.chat.id,
      '/start - Start bot\n/help - Show help\nI will echo your messages.',
    );
  });

  bot.on('message', (msg) => {
    if (!msg.text) {
      return;
    }

    if (msg.text.startsWith('/')) {
      return;
    }

    bot.sendMessage(msg.chat.id, `You said: ${msg.text}`);
  });

  console.log('Bot is running...');
};

main();

File: Python/Bot.py
import os

from telegram import Update
from telegram.ext import (
    ApplicationBuilder,
    CommandHandler,
    ContextTypes,
    MessageHandler,
    filters,
)


async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(
        "Hello! I am your simple bot.\nUse /help to see commands."
    )


async def help_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(
        "/start - Start bot\n/help - Show help\nI will echo your messages."
    )


async def echo(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    if update.message and update.message.text:
        await update.message.reply_text(f"You said: {update.message.text}")


def main() -> None:
    token = os.getenv("TELEGRAM_BOT_TOKEN")
    if not token:
        print("Error: TELEGRAM_BOT_TOKEN is not set.")
        return

    app = ApplicationBuilder().token(token).build()

    app.add_handler(CommandHandler("start", start))
    app.add_handler(CommandHandler("help", help_command))
    app.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, echo))

    print("Bot is running...")
    app.run_polling()


if __name__ == "__main__":
    main()
